core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -499,29 +503,29 @@ private[spark] class ExecutorAllocationManager(
    */
   private[spark] class ExecutorAllocationListener extends SparkListener {
 
-    private val stageIdToNumTasks = new mutable.HashMap[Int, Int]
-    // Number of running tasks per stage including speculative tasks.
-    // Should be 0 when no stages are active.
-    private val stageIdToNumRunningTask = new mutable.HashMap[Int, Int]
-    private val stageIdToTaskIndices = new mutable.HashMap[Int, mutable.HashSet[Int]]
-    // Number of speculative tasks to be scheduled in each stage
-    private val stageIdToNumSpeculativeTasks = new mutable.HashMap[Int, Int]
-    // The speculative tasks started in each stage
-    private val stageIdToSpeculativeTaskIndices = new mutable.HashMap[Int, mutable.HashSet[Int]]
-
-    // stageId to tuple (the number of task with locality preferences, a map where each pair is a
-    // node and the number of tasks that would like to be scheduled on that node) map,
-    // maintain the executor placement hints for each stage Id used by resource framework to better
-    // place the executors.
-    private val stageIdToExecutorPlacementHints = new mutable.HashMap[Int, (Int, Map[String, Int])]
-
+    private val stageAttemptToNumTasks = new mutable.HashMap[StageAttempt, Int]
+    private val stageAttemptToTaskIndices = new mutable.HashMap[StageAttempt, mutable.HashSet[Int]]
+    // Number of speculative tasks to be scheduled in each stageAttempt
+    private val stageAttemptToNumSpeculativeTasks = new mutable.HashMap[StageAttempt, Int]
+    // The speculative tasks started in each stageAttempt
+    private val stageAttemptToSpeculativeTaskIndices =
+      new mutable.HashMap[StageAttempt, mutable.HashSet[Int]]
+    private val liveTaskIds = new mutable.HashSet[Long]
```

Comment:
I'm still in doubt whether we shall keep track of all the running taskIds here, especially when the only use case can be covered by `stageIdToNumRunningTask.values.sum`

Reply:
The previous commit version used ```executorIdToTaskIds.values.map(_.size).sum```, but in [SPARK-20286](https://issues.apache.org/jira/browse/SPARK-20286), ```executorIdToTaskIds``` was removed.

It seems that recording taskid is not important.
Now I changed ```stageIdToNumRunningTask``` to ```stageAttemptToNumRunningTask```.